akka-sample-fsm-java/src/main/java/sample/Chopstick.java

```diff
@@ -0,0 +1,125 @@
+package sample;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+
+public class Chopstick {
+    interface ChopstickMessage { }
+
+    final static class Take implements ChopstickMessage {
+        public final ActorRef<ChopstickAnswer> hakker;
+
+        public Take(ActorRef<ChopstickAnswer> hakker) {
+            this.hakker = hakker;
+        }
+    }
+
+    final static class Put implements ChopstickMessage {
+        public final ActorRef<ChopstickAnswer> hakker;
+
+        public Put(ActorRef<ChopstickAnswer> hakker) {
+            this.hakker = hakker;
+        }
+    }
+
+    public static Behavior<ChopstickMessage> create() {
+        return Behaviors.setup(context-> new Chopstick(context).available());
+    }
+
+    interface ChopstickAnswer {
+
+        ActorRef<ChopstickMessage> getChopstick();
+
+        default boolean isTakenBy() {
+            return false;
+        }
+
+        default boolean isTakenBy(ActorRef<ChopstickMessage> chopstick) {
+            return false;
+        }
+
+        default boolean isBusy() {
+            return false;
+        }
+
+        default boolean isBusy(ActorRef<ChopstickMessage> chopstick) {
+            return false;
+        }
```

Comment:
What this mean `answer.isBusy(chopstick)` that it is busy `for` that chopstick?

Reply:
Yeah that is rather weird (taken from existing code). I'll just compare the chopstick in the message to the chopstick the hakker wants